wallet/src/main/java/com/android/identity_credential/wallet/ui/destination/document/ActivityLogScreen.kt

```diff
@@ -0,0 +1,107 @@
+package com.android.identity_credential.wallet.ui.destination.document
+
+import androidx.compose.foundation.ExperimentalFoundationApi
+import androidx.compose.foundation.layout.Arrangement
+import androidx.compose.foundation.layout.Box
+import androidx.compose.foundation.layout.Column
+import androidx.compose.foundation.layout.Row
+import androidx.compose.foundation.layout.fillMaxHeight
+import androidx.compose.foundation.layout.fillMaxSize
+import androidx.compose.foundation.layout.fillMaxWidth
+import androidx.compose.foundation.layout.heightIn
+import androidx.compose.foundation.layout.padding
+import androidx.compose.foundation.lazy.LazyColumn
+import androidx.compose.foundation.lazy.items
+import androidx.compose.material3.Button
+import androidx.compose.material3.MaterialTheme
+import androidx.compose.material3.Text
+import androidx.compose.runtime.Composable
+import androidx.compose.runtime.mutableStateListOf
+import androidx.compose.runtime.mutableStateOf
+import androidx.compose.runtime.remember
+import androidx.compose.ui.Alignment
+import androidx.compose.ui.Modifier
+import androidx.compose.ui.res.stringResource
+import androidx.compose.ui.text.style.TextAlign
+import androidx.compose.ui.unit.dp
+import com.android.identity_credential.wallet.R
+import com.android.identity_credential.wallet.navigation.WalletDestination
+import com.android.identity_credential.wallet.ui.KeyValuePairText
+import com.android.identity_credential.wallet.ui.ScreenWithAppBarAndBackButton
+import com.android.identity_credential.wallet.util.ActivityEntry
+import com.android.identity_credential.wallet.util.ActivityLogger
+
+private const val TAG = "ActivitiesLogScreen"
+
+@OptIn(ExperimentalFoundationApi::class)
+@Composable
+fun ActivityLogScreen(documentId: String, onNavigate: (String) -> Unit) {
+    val entries = remember { mutableStateListOf<ActivityEntry>().apply { addAll(ActivityLogger.getEntries(documentId)) } }
```

Comment:
Similar to `DocumentInfoScreen` and `DocumentDetailsScreen` I think you should use a method on `DocumentModel` to generate objects suitable for displaying in the UI. So something like

```
    val eventInfos = documentModel.getEventInfos(documentId)
```
where `DocumentModel.getEventInfos()` would be return `List<EventInfo`> where each `EventInfo` would contain things like `String` and `Bitmap`, not things like `ByteArray` or other stuff that we store in the log. More below about that...

Reply:
documentModel.getEventInfos is done. The EventInfo itself contains timestamp, thumbnail but I kept the ByteArray for now since I ma now what would you expect the Bitmap to contain (QR code?). It appears that you wanted to add more comment but missed that?